egs/librispeech/ASR/conformer_ctc/asr_datamodule.py

```diff
@@ -230,21 +300,63 @@ def test_dataloaders(self) -> Union[DataLoader, List[DataLoader]]:
         for cuts_test in cuts:
             logging.debug("About to create test dataset")
             test = K2SpeechRecognitionDataset(
-                cuts_test,
                 input_strategy=OnTheFlyFeatures(
-                    Fbank(FbankConfig(num_mel_bins=80))
+                    Fbank(FbankConfig(num_mel_bins=80), num_workers=4)
+                    if self.args.on_the_fly_feats
+                    else PrecomputedFeatures()
                 ),
```

Comment:
I think this closing parenthesis has to be moved 3 lines up, so the code looks like:

```python
input_strategy=OnTheFlyFeatures(
   Fbank(FbankConfig(num_mel_bins=80), num_workers=4)
)  if self.args.on_the_fly_feats
   else PrecomputedFeatures(),
return_cuts=...
```

Currently when args.on_the_fly_feats = False, it tries to use OnTheFlyFeatures(PrecomputedFeatures()) which is an error.

Reply:
Thanks!
In that case, I think we should also change snowfall to fix that as this block of code is from snowfall.
See
https://github.com/k2-fsa/snowfall/blob/1f79957e9716c3f980c151df5b1d77bc4bb7ce78/egs/gigaspeech/asr/simple_v1/asr_datamodule.py#L337-L344
```python
            test = K2SpeechRecognitionDataset(
                input_strategy=(
                    OnTheFlyFeatures(Fbank(FbankConfig(num_mel_bins=80)), num_workers=8)
                    if self.args.on_the_fly_feats
                    else PrecomputedFeatures()
                ),
                return_cuts=self.args.return_cuts,
            )
```

Reply:
yes, you’re right